[yuhinarita] iP

yuhinarita opened the pull request:
# ChipChat
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

ChipChat frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn  
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is ***FREE***!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yuhinarita committed:
Level 0. Rename, Greet, Exit

yuhinarita committed:
Level 1. Echo

yuhinarita committed:
Level 2. Add, List

yuhinarita committed:
Level 3. Mark as Done

yuhinarita committed:
Level 4. ToDos, Events, Deadlines

yuhinarita committed:
Duke A-TextUiTesting: Automated Text UI Testing

yuhinarita committed:
Level 5 and level 6

yuhinarita committed:
Rename chatbot to Chipchat

yuhinarita committed:
Add save and load to local file feature

yuhinarita committed:
Merge branch 'branch-Level-7'

yuhinarita committed:
Level-8 Dates

yuhinarita committed:
Modify to be more OOP & turn into package

yuhinarita committed:
Turn into package

yuhinarita committed:
Fix naming for addTask methods

yuhinarita committed:
Add argument exceptions

yuhinarita committed:
Fix switch-case indentation to follow coding standard

Colex2000 commented on src/main/java/chipchat/action/AddTask.java:
```diff
@@ -0,0 +1,48 @@
+
+import java.time.LocalDate;
+
+public class AddTask extends Action {
```
Perhaps can change the name to sound more "noun" instead. It sounds more verbish right now 😆. Can check here!
https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming

Colex2000 commented on src/main/java/chipchat/action/AddTask.java:
```diff
@@ -0,0 +1,48 @@
+        this.taskType = taskType;
+    }
+
+    public static Action addTodo(String description, boolean isDone) {
```
Naming style for these methods sounds GOOD! Awesome :)

Colex2000 commented on src/main/java/chipchat/action/CommandType.java:
```diff
@@ -0,0 +1,20 @@
+
+    public static CommandType matchTaskType(String str) throws ChipchatException {
+        switch(str.toUpperCase()){
+            case "T":
```
I THINK for our mod, we don't have indentation for switch and case hahaha. You can check the coding style for this mod again!

Colex2000 reviewed (commented):
Overall, very decent naming for classes, methods, and variables. Good job! Just some minor comments.

yuhinarita committed:
Modify parser

yuhinarita committed:
Modify logic to parse saved data

yuhinarita committed:
Merge branch 'add-gradle-support'

yuhinarita committed:
Add junit testcases for Parser

yuhinarita committed:
Add JUnit testcases for TaskList

xinen26 commented on src/main/java/chipchat/action/CommandType.java:
```diff
@@ -0,0 +1,7 @@
+
+import chipchat.exception.ChipchatException;
+
+public enum CommandType {
```
I like how you managed to incorporate enums into your code!

xinen26 commented on src/main/java/chipchat/action/Unmark.java:
```diff
@@ -0,0 +1,25 @@
+import chipchat.ui.Ui;
+
+public class Unmark extends Action {
+    private final String SUCCESS_MSG = "Nice! I've unmarked this task as not done:";
```
i like how you saved your string message as a constant variable! i havent thought about it before but now will try to incorporate into my ip too!

xinen26 commented on src/main/java/chipchat/parser/Parser.java:
```diff
@@ -0,0 +1,140 @@
+
+        int index = parseIndex(tokens[1]);
+        switch(command) {
+            case DELETE:
```
just wanna remind that there should be no indentation for case clauses, but overall, I really like how neat and tidy this chunk of code is

xinen26 commented on src/main/java/chipchat/parser/Parser.java:
```diff
@@ -0,0 +1,140 @@
+        ArrayList<String> args = parseArguments(tokens);
+        String description = args.get(0);
+        switch(command) {
+            case TODO:
```
i think there should be no indentation for case clauses

xinen26 commented on src/test/java/chipchat/parser/ParserTest.java:
```diff
@@ -0,0 +1,69 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ParserTest {
```
your ParserTest is really comprehensive and takes into account most, if not all, test cases and i think that's great!

xinen26 commented on src/test/java/chipchat/parser/ParserTest.java:
```diff
@@ -0,0 +1,69 @@
+    public void parseAction_feedUserInput_returnCorrectAction() {
+        Action action;
+
+        action = Parser.parseAction("bye");
```
i really like how neat and tidy this code is, will definitely re-implement/tidy up my own test classes to be neater with inspiration from your test class

xinen26 reviewed (commented):
I think your code is really nice, neat and creative! Reviewing your code definitely gave me some new ideas on to improve my IP! :)

yuhinarita committed:
Add javadocs

yuhinarita committed:
Fix coding standard

yuhinarita committed:
Add find task feature

yuhinarita committed:
Add checkstyle

yuhinarita committed:
Remove old main class

yuhinarita committed:
Fix checkstyle warnings

yuhinarita committed:
Add final keyword on all fields

yuhinarita committed:
Include javaFX dependencies in gradle

yuhinarita committed:
Add javaFX

yuhinarita committed:
Integrate code to use JavaFx as output

yuhinarita committed:
Fix issue of output text being truncated

yuhinarita committed:
Add assertion that command and index is not empty

yuhinarita committed:
Add Chipchat.jar

yuhinarita committed:
Add assertion that command and index is not empty

yuhinarita committed:
Throw exception at default branch of switch

The switch statement returns null at default branch.

This is needs to be handled properly with future extensions in mind.

It will now throw an ArgumentException indicating that the null branch
has been reached in which method.

yuhinarita committed:
Merge pull request #3 from yuhinarita/branch-A-Assertions

Add assertion that command and index is not empty

yuhinarita committed:
Merge pull request #4 from yuhinarita/branch-A-CodeQuality

Throw exception at default branch of switch

yuhinarita committed:
Add feature so tasks can have tags

This will help the user categorize tasks based on assigned tags.

Currently, it can only be assigned when adding a new task, and cannot be
searched by tag. In future, hopefuly we can extend the search feature to also look for tags.

yuhinarita committed:
Merge pull request #5 from yuhinarita/branch-BCD-Extension

Add feature so tasks can have tags

yuhinarita committed:
Fix parse method so tagging is optional

yuhinarita committed:
Remove unused import

yuhinarita committed:
Show greetings message at startup

yuhinarita committed:
Fix parsing method for loading saved data

yuhinarita committed:
Fix coding quality

yuhinarita committed:
Refactor parseLoadTask() to reuse parseTask()

yuhinarita committed:
Remove unused parseArguments() method

yuhinarita committed:
Remove unused imports

yuhinarita committed:
Remove unused import

yuhinarita committed:
Update javadocs

yuhinarita committed:
Add title bar to GUI

yuhinarita committed:
Fix bug where same tag was loaded

yuhinarita committed:
Update README.md

yuhinarita committed:
Update README.md

yuhinarita committed:
Merge branch 'master' of https://github.com/yuhinarita/ip

yuhinarita committed:
Refactor parser

yuhinarita committed:
Update javadocs

yuhinarita committed:
Update README.md

yuhinarita committed:
Merge branch 'master' of https://github.com/yuhinarita/ip

yuhinarita committed:
Fix storage issues

yuhinarita committed:
Update README.md

yuhinarita committed:
Update README.md

yuhinarita committed:
Update README.md

yuhinarita committed:
Update README.md

yuhinarita committed:
Change storage path

yuhinarita committed:
Merge branch 'master' of https://github.com/yuhinarita/ip

yuhinarita committed:
Update README.md

yuhinarita committed:
Update README.md

yuhinarita committed:
Add Ui.png

yuhinarita committed:
Merge branch 'master' of https://github.com/yuhinarita/ip

yuhinarita committed:
Add Ui.png
